arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/ModuleOperationStackedSet.java

```diff
@@ -134,11 +146,25 @@ public void clear() {
 
   public void finishConflation() {
     conflationFinished = true;
+    if (!duplicateAlreadyRemoved) {
+      lineCount();
+    }
     operationsCommitedToTheConflation().addAll(operationsInTransaction());
     operationsInTransaction().clear();
     lineCounter.enter(); // this is not mandatory but it is more consistent
   }
 
+  void deleteDuplicates() {
+    final List<E> toRemove =
+        new ArrayList<>(operationsInTransaction().size() * EXPECTED_PROPORTION_OF_DUPLICATE);
+    for (E e : operationsInTransaction()) {
+      if (operationsCommitedToTheConflation().contains(e)) {
```

Comment:
This is what we need to assess in terms of performance. If I understand correctly `deleteDuplicates()` is called each time on transaction `n-1` when we enter a new transaction `n` or when the conflation finishes. The difference from the previous implementation is that with this implementation, we already know that the operation is unique for transaction n-1, where with the previous implementation, we can still check duplicates from transaction n-1 multiple times towards the conflated traces.
One other solution that is very close from current implementation (before this PR) is to check on `add` method if the operation is already present in the current transaction (`operationsInTransaction`), if it is the case, we do nothing, otherwise, we need to check if it is present in `operationsCommitedToTheConflation`. If it is there, we don't need to add it the the current transaction (`operationsInTransaction`), otherwise we add it to `operationsInTransaction`.